infrastructure/async/src/test/java/tech/pegasys/teku/infrastructure/async/SafeFutureTest.java

```diff
@@ -441,6 +441,33 @@ public void propagateTo_propagatesExceptionalResult() {
     assertThatSafeFuture(target).isCompletedExceptionallyWith(exception);
   }
 
+  @Test
+  public void propagateToAsync_propagatesSuccessfulResult() {
+    final StubAsyncRunner asyncRunner = new StubAsyncRunner();
+    final SafeFuture<String> target = new SafeFuture<>();
+    final SafeFuture<String> source = new SafeFuture<>();
+    source.propagateToAsync(target, asyncRunner);
+    source.complete("Yay");
+    assertThat(target).isNotDone();
+
+    asyncRunner.executeQueuedActions();
+    assertThat(target).isCompletedWithValue("Yay");
+  }
+
+  @Test
+  public void propagateToAsync_propagatesExceptionalResult() {
+    final StubAsyncRunner asyncRunner = new StubAsyncRunner();
+    final SafeFuture<String> target = new SafeFuture<>();
+    final SafeFuture<String> source = new SafeFuture<>();
+    source.propagateToAsync(target, asyncRunner);
+    final RuntimeException exception = new RuntimeException("Oh no!");
+    source.completeExceptionally(exception);
+    assertThat(target).isNotDone();
+
+    asyncRunner.executeQueuedActions();
+    assertThatSafeFuture(target).isCompletedExceptionallyWith(exception);
+  }
```

Comment:
Somehow these didn't make it into #2362.  Should be the only actual code changes in here (the rest being build files and imports changing).